DREAM3DReviewFilters/FindArrayStatistics.cpp

```diff
@@ -434,6 +462,74 @@ double findSummation(C<T, Ts...>& source)
   return sum;
 }
 
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+template <template <typename, typename...> class C, typename T, typename... Ts>
+std::vector<float> findHistogram(C<T, Ts...>& source, float histmin, float histmax, bool histfullrange, int32_t numBins)
+{
+
+  std::vector<float> Histogram(numBins, 0);
+  int overflow = 0;
+  int32_t bin = 0;
+  int32_t numPoints = source.size();
+
+  if(source.empty())
+  {
+    std::vector<float> empty(numBins, 0);
+    return empty;
+  }
+
+  float min = std::numeric_limits<float>::max();
+  float max = -1.0 * std::numeric_limits<float>::max();
+
+  if(histfullrange)
+  {
+    min = findMin(source);
+    max = findMax(source);
+  }
+  else
+  {
+    min = histmin;
+    max = histmax;
+  }
+
+  float increment = (max - min) / (numBins);
+  if(abs(increment) < 1E-10)
+  {
+    numBins = 1;
+  }
+
+  if(numBins == 1) // if one bin, just set the first element to total number of points
+  {
+    Histogram[0] = static_cast<float>(source.size());
+  }
+  else
+  {
+    for(const auto& s : source)
+    {
+      bin = size_t((s - min) / increment); // find bin for this input array value
```

Comment:
```suggestion
      size_t bin = static_cast<size_t>((s - min) / increment); // find bin for this input array value
```
Can move the declaration of bin inside the loop and changed it to be size_t.
Change C style cast to static_cast.